lib/smart_proxy_reports/spooled_http_client.rb

```diff
@@ -52,37 +52,44 @@ def stop
       @worker.join
     end
 
+    def connect(http, factory, type, endpoint)
+      new = 0
```

Comment:
Generally it is better to avoid things that are keywords in other languages. In Ruby `new` is probably allowed, but I would suggest to avoid these.